### ISSUE TITLE
Response Time

### ISSUE THREAD

**Issue by albertz** [openlierox/openlierox#241]
Converted from [SourceForge issue 2785029](http://sourceforge.net/support/tracker.php?aid=2785029), submitted by daevil1

All beta 9 exe's I've tried have given me the same issue. When I start playing a game it works fine, and it can last for a while. Though after a while (could be 1 minute of playing, or it could be 20 minutes) I get a delayed response time for my worm. If I click left, it will take 1-2 seconds for it to move left. Same with all other controls. The only way I've been able to solve this, is to jump back to Windows Vista, switch to another task, and do some random stuff there, and after a few tries response time is what it should be again. The last version of OLX I tried this with was a beta I downloaded on the 26th april.

**Comment by albertz** [comment]
Submitted by albertzeyer

Can you post the full console output (from start to the time where that problem happens)?

Which mod are you playing? Is this local play or network? Are you hosting or have you joined another server?

Can you give some other details about your computer, like CPU?

**Comment by albertz** [comment]
Submitted by daevil1

Unfortunatly (or fortunatly) I haven't been able to replicate the bug, so I don't have an stdout file yet to help describe the problem.

CPU: 2.20 Ghz (two cores) laptop, 2.00 GB ram Nvidia Geforce 8400M GS 256-1023 MB (shared system memory of 767mb)

Usually I've been playing on other servers when the bug has occoured, and most of the time 100lt Liero 1.0. 

I'm not sure, but I think it has happened in local play also with liero 1.0.

**Comment by albertz** [comment]
Submitted by albertzeyer

If you got one (console output), would be nice to have it. :)

What is your MaxFPS setting? Is it set to 0? (Which means max possible FPS.) Or some value?

Also, can you take a look at a system monitor (or task manager or however that is called under Windows) and watch the CPU usage by OpenLieroX? In case you have MaxFPS=0, it should be 100%, or if Windows count it like Linux, it should be almost 200%, because OLX is highly multithreaded. The case MaxFPS>0 is more interesting. I wonder if the CPU usage gets noteable higher when the bug occurs.

Do you have low FPS when the problem occurs?

And once the problem occurs, does it persists until the quit of OLX? Or is it gone again in the next round?

**Comment by albertz** [comment]
Submitted by daevil1

I'm trying to replicate the bug to get a console output, so if it happends, I'll upload it.

MaxFPS seemed to be set on 200. that used under 50% of resources. When I set MaxFPS to 0 it stayed between 200 and 300 depending on objects (with 5 game over shots in 8.bit warfare 1.2 i got 97 fps at lowest.) The CPU usage was then up at about 100% And I'm quite sure it can't be displayed as 200% in Windows. But I'm not too familiair with how Linux works though.

I didn't check CPU usage when the bug occoured, but the PC seemed to run just about the same when the bug was occouring as opposed to when it wasn't. I do not remember a drop in FPS when the problem occoured. Just the delayed response time.

If it's of interest, one time when I got the bug while playing online, I tried to make it go away, by tabbing out and doing some tasks in the background, then when I came back it was fixed, and above my worm, it said "away". When I got the "away" thing to go away by tabbing out and tabbing back in, the bug came right back. When I managed to get the bug away again, I got the "away" text above my worm again. This only happened in one OLX session. When I restarted it, I didn't get that again.

When I get the problem, it does not go away between rounds. Sometimes it seems to dissappear at random moments, or if I tab out a few times and tab back into the game. But other than that, I also get the bug when in the lobby as well. It has only started once I've been in the game. But when the round is finished, and we're back in the lobby, if I try to write something in the textbox, it has the same delay that the worm has.

**Comment by albertz** [comment]
Submitted by albertzeyer

(Well, on Linux it very much depends on the tool. Some of them show %CPU-usage on system, some/most show %CPU-usage per core. With multiple cores, this is of course not the same.)

So you are saying that in some situations, it showed permanently "away" over your worm but you have not been away? This was another bug, which should be fixed already (quite a while ago). If you see this again, please make another bug report about that.

The bug seems strange to me. After your description, I got somehow the impression that it's like a video and the frames got stuck and delayed somehow (for about a second). Like if OLX sends a lot of video updates and Vista tries to show each single frame but is slightly to slow. AFAIK, Vista works somehow like that (they called it managed window drawing or something like that; it is new since Vista). (Most modern OS work quite a while like that, i.e. without DirectX/OpenGL, they never can draw directly on the screen.) Would that make sense somehow to you? Could that be?

Have you tried with OpenGL activated? (Or is it already activated?) Do you have the problem also in fullscreen?

Btw., when it shows "away" over your worm, that means that OLX thinks that it is in the background right now. In that case, it slows down the FPS. That would also fit to my theory, because it is likely that Vista can show a lower FPS fast enough. You could perhaps also try out to set a lower MaxFPS, like MaxFPS=50 or so. Is there any difference for you with such MaxFPS?

Karel, do you know some more details about Vista, if it could be like I described?

Btw., I know some earlier revisions where we had a similar bug (though not exactly the same as you described). But you said you had also tried rev 4180 and the bug should not be there anymore.

**Comment by albertz** [comment]
Submitted by daevil1

Yeah, it showed permanantly "away" even when I wasn't one time. But I think that was with an earlier revision. I haven't experienced it after though.

I don't think it's a problem with the frames. I think it was an issue with delayed response from the keyboard. I have DirectX10. And DirectDraw acceleration, Direct3D acceleration and AGP Texture Acceleration are all enabled.

I don't think I have OpenGL, but if I get the problem again, I could try to install OpenGL and see if that makes a difference.

When I set MaxFPS to 50, it didn't seem too different. It runs a little less smooth, but the FPS is very stable, and the system uses about 40% of CPU with a few spikes up and down. Other than that, it was as usual.

What was the exact bug in earlier revisions? Was it FPS related, or was it something else?

**Comment by albertz** [comment]
Submitted by dark_charlie

OLX uses DirectX by default, that means direct drawing on the screen. There cannot be that huge delays even with delayed drawing though.

OpenGL is installed in Windows Vista, you need to only tick the option in OLX. It is usually slower than the classic rendering but sometimes fixes graphics problems.

The bug in previous versions was the incorrect detection of the away status.

**Comment by albertz** [comment]
Submitted by albertzeyer

The old problem (which should be fixed though) was on systems with unlimited (or very high) MaxFPS where the drawing was much slower than all the rest. If it is that problem, chances are very good that it is gone (any delay) with some low MaxFPS. That's why I am asking if you can reproduce the problem with low MaxFPS.

"Other than that, it was as usual." means that you have also the delay/problem with MaxFPS=50 ?

**Comment by albertz** [comment]
Submitted by daevil1

No, I have no delay problem with maxFPS=50

I tried with OpenGL both with MaxFPS=50 and MaxFPS=0 and that was more or less the same as with DirectDraw (OpenGL) So so far I have no luck with reproducing the problem with low FPS. Could it be IRQ related perhaps? Sounds far fetched to me though...

**Comment by albertz** [comment]
Submitted by albertzeyer

Ok, being a bit more technical: I think the problem is, with high FPS, we put too much video update request events into the SDL event queue. Or we lock it too often. Not exactly sure on that (esp because I already made a lot to prevent such cases, and because I thought I had fixed that already).

**Comment by albertz** [comment]
Submitted by albertzeyer

Can you retry with most recent EXE, please?

**Comment by albertz** [comment]
Submitted by nobody

I tried a quick game against the computer, and the bug did not show. I'll try online later, and report back then

**Comment by albertz** [comment]
Submitted by sf-robot

This Tracker item was closed automatically by the system. It was
previously set to a Pending status, and the original submitter
did not respond within 14 days (the time period specified by
the administrator of this Tracker).
